### DISCUSSION TITLE
Something like «makemigrations»

### DISCUSSION THREAD

**Discussion by hoosnick** [piccolo-orm/piccolo#861] [General]
How can we implement the `makemigrations` command like in [Django](https://docs.djangoproject.com/en/4.2/ref/django-admin/#django-admin-makemigrations)? 
Why is it not implemented? It is not convenient to manually create migrations for each application and then migrate.

**Comment by dantownsend** [comment]
Hi @hoosnick. Piccolo has this:

```
piccolo migrations new my_app --auto
```

Which is basically the same as how Django works (it looks at the tables registered in your `piccolo_app.py` file, compares them to existing migrations, and automatically creates a new migration file containing the differences).

I think the docs need reordering, as it's at the bottom of the page when it should be the first thing really.

**Reply by hoosnick** [reply]
https://github.com/piccolo-orm/piccolo/issues/862#issuecomment-1633911634
